dbms/src/Storages/KVStore/MultiRaft/PrehandleSnapshot.cpp

```diff
@@ -163,6 +200,328 @@ PrehandleResult KVStore::preHandleSnapshotToFiles(
+    ReadFromStreamResult & result,
+    PrehandleResult & prehandle_result,
+    const std::vector<std::string> & split_keys,
+    std::shared_ptr<DM::SSTFilesToBlockInputStream> sst_stream,
+    RegionPtr new_region,
+    DM::SSTFilesToBlockInputStreamOpts & opt,
+    const SSTViewVec & snaps,
+    const TiFlashRaftProxyHelper * proxy_helper,
+    TMTContext & tmt,
+    std::shared_ptr<std::atomic_bool> prehandle_task,
+    DM::FileConvertJobType job_type,
+    uint64_t index,
+    std::shared_ptr<StorageDeltaMerge> storage)
+{
+    using SingleSnapshotAsyncTasks = AsyncTasks<uint64_t, std::function<bool()>, bool>;
+    auto split_key_count = split_keys.size();
+    RUNTIME_CHECK_MSG(
+        split_key_count >= 1,
+        "split_key_count should be more or equal than 1, actual {}",
+        split_key_count);
+    LOG_INFO(
+        log,
+        "Parallel prehandling for single big region, range={}, split keys={}, region_id={}",
+        new_region->getRange()->toDebugString(),
+        split_key_count,
+        new_region->id());
+    auto async_tasks = SingleSnapshotAsyncTasks(split_key_count, split_key_count, split_key_count + 5);
+    sst_stream->resetSoftLimit(
+        DM::SSTScanSoftLimit(DM::SSTScanSoftLimit::HEAD_OR_ONLY_SPLIT, std::string(""), std::string(split_keys[0])));
+
+    ParallelPrehandleCtxPtr ctx = std::make_shared<ParallelPrehandleCtx>();
+
+    for (size_t extra_id = 0; extra_id < split_key_count; extra_id++)
+    {
+        auto add_result = async_tasks.addTask(extra_id, [&, extra_id]() {
+            auto limit = DM::SSTScanSoftLimit(
+                extra_id,
+                std::string(split_keys[extra_id]),
+                extra_id + 1 == split_key_count ? std::string("") : std::string(split_keys[extra_id + 1]));
+            LOG_INFO(
+                log,
+                "Add extra parallel prehandle task split_id={}/total={} limit {}",
+                extra_id,
+                split_keys.size(),
+                limit.toDebugString());
+            runInParallel(
+                log,
+                new_region,
+                opt,
+                snaps,
+                proxy_helper,
+                tmt,
+                prehandle_task,
+                job_type,
+                index,
+                extra_id,
+                ctx,
+                std::move(limit),
+                storage);
+            return true;
+        });
+        RUNTIME_CHECK_MSG(
+            add_result,
+            "Failed when add {}-th task for prehandling region_id={}",
+            extra_id,
+            new_region->id());
+    }
+    LOG_INFO(
+        log,
+        "Add extra parallel prehandle task split_id={}/total={} limit {}",
+        DM::SSTScanSoftLimit::HEAD_OR_ONLY_SPLIT,
+        split_keys.size(),
+        sst_stream->getSoftLimit().value().toDebugString());
+    auto [head_result, head_prehandle_result]
+        = executeTransform(new_region, prehandle_task, job_type, storage, sst_stream, opt, tmt);
```

Comment:
why we still need adding this?

Reply:
I will move these logs into executeTransform